src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,214 @@
+          messageId: 'preferImportingJestGlobal',
+          data: { jestFunctions: jestFunctionsToImportFormatted },
+          fix(fixer) {
+            const sourceCode = getSourceCode(context);
+            const [firstNode] = sourceCode.ast.body;
+
+            // check if "use strict" directive exists
+            if (
+              firstNode.type === AST_NODE_TYPES.ExpressionStatement &&
+              firstNode.expression.type === AST_NODE_TYPES.Literal &&
+              firstNode.expression.value === 'use strict'
+            ) {
+              return fixer.insertTextAfter(
+                firstNode,
+                `\n${createFixerImports(isModule, jestFunctionsToImport)}`,
+              );
+            }
+
+            const importNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.ImportDeclaration &&
+                node.source.value === '@jest/globals',
+            );
+
+            if (
+              importNode &&
+              importNode.type === AST_NODE_TYPES.ImportDeclaration
+            ) {
+              const existingImports = importNode.specifiers.reduce<string[]>(
+                (imports, specifier) => {
+                  if (
+                    specifier.type === AST_NODE_TYPES.ImportSpecifier &&
+                    specifier.imported?.name
+                  ) {
+                    imports.push(specifier.imported.name);
+                  }
+
+                  if (
+                    specifier.type === AST_NODE_TYPES.ImportDefaultSpecifier
+                  ) {
+                    imports.push(specifier.local.name);
+                  }
+
+                  return imports;
+                },
+                [],
+              );
+
+              const allImports = [
+                ...new Set([...existingImports, ...jestFunctionsToImport]),
+              ];
+
+              return fixer.replaceText(
+                importNode,
+                createFixerImports(isModule, allImports),
+              );
+            }
+
+            const requireNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.VariableDeclaration &&
+                node.declarations.some(
+                  declaration =>
+                    declaration.init &&
+                    declaration.init.type === AST_NODE_TYPES.CallExpression &&
+                    declaration.init.callee.type ===
+                      AST_NODE_TYPES.Identifier &&
+                    declaration.init.callee.name === 'require' &&
+                    declaration.init.arguments[0]?.type === 'Literal' &&
+                    declaration.init.arguments[0]?.value === '@jest/globals' &&
+                    (declaration.id.type === AST_NODE_TYPES.Identifier ||
+                      declaration.id.type === AST_NODE_TYPES.ObjectPattern),
+                ),
+            );
+
+            if (requireNode?.type === AST_NODE_TYPES.VariableDeclaration) {
```

Comment:
I'd prefer to do an early return for the other branch since its a lot shorter

Reply:
modified